packages/lib-sourcify/src/Compilation/auxdataUtils.ts

```diff
@@ -88,56 +88,75 @@ function getAuxdatasDiff(originalAuxdatas: string[], editedAuxdatas: string[]) {
 }
 
 /**
- * Finds the positions of the auxdata in the bytecode.
- * The compiler outputs the auxdata values in the `legacyAssembly` field. However we can't use these values to do a simple string search on the compiled bytecode because an attacker can embed these values in the compiled contract code and cause the correspoding field in the onchain bytecode to be ignored falsely during verification.
- * A way to find the *metadata hashes* in the bytecode is to recompile the contract with a slightly edited source code and compare the differences in the raw bytecodes. However, this will only give us the positions of the metadata hashes in the bytecode. We need to find the positions of the whole *auxdata* in the bytecode.
- * So we go through each of the differences in the raw bytecode and check if an auxdata diff value from the legacyAssembly is included in that difference. If it is, we have found the position of the auxdata in the bytecode.
+ * Maps every legacy-assembly auxdata block to its `{ offset, value }`
+ * inside the recompiled bytecode.
+ *
+ * @param originalBytecode – bytecode produced from the **original** sources
+ * @param editedBytecode   – bytecode produced from the *slightly different* sources
+ * @param originalAuxdatas – auxdata strings coming from the original compile
+ * @param editedAuxdatas   – auxdata strings coming from the edited compile
+ *
+ * @returns an object whose keys are the indexes of the auxdata items
+ *          (as they appear in `legacyAssembly`) and whose values are
+ *          their position and value in the originalBytecode.
  */
 export function findAuxdataPositions(
   originalBytecode: string,
   editedBytecode: string,
   originalAuxdatas: string[],
   editedAuxdatas: string[],
 ): CompiledContractCborAuxdata {
-  const auxdataDiffObjects = getAuxdatasDiff(originalAuxdatas, editedAuxdatas);
+  // 1. Build auxdataDiffs that explain how each auxdata is changing after recompilation with slightly different sources
 
-  const diffPositionsBytecodes = getDiffPositions(
+  // A single AuxdataDiff has 3 keys: real, diffStart (int offset of diff), diff
+  // original: a2646970667358221220123af2412fd4b1b89740f76c6ab76a6412bbde98fab732fb97eb012abe7123ff64736f6c63430007000033
+  //           └────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+  //               REAL
+  // edited:   a2646970667358221220dceca8706b29e917dacf25fceef95acac8d90d765ac926663ce4096195952b6164736f6c63430007000033
+  //           └──────────────────┘└──────────────────────────────────────────────────────────────┘
+  //               DIFFSTART           DIFF
+  const auxdataDiffs = getAuxdatasDiff(originalAuxdatas, editedAuxdatas);
```

Comment:
In line with the suggestion above to change param names
```suggestion
  const compilerAuxdataDiffs = getAuxdatasDiff(originalAuxdatas, editedAuxdatas);
```